### PR TITLE
Added missing this in math tests

### PR DESCRIPTION
**Pull request by davidpiuva**
Added lots and lots of missing `this`. Starting with the math folders.

### CODE REVIEW

**Comment by davidpiuva** [comment]
@fredrikbryntesson Peer review?

File: test/math/FloatComplexVectorListTest.ooc [hunk 1]
```diff
@@ -19,39 +19,39 @@ FloatComplexVectorListTest: class extends Fixture {
...
 
 	complexNumberArray := FloatComplexVectorList new(4)
-	complexNumberArray add(complexNumber0)
-	complexNumberArray add(complexNumber1)
-	complexNumberArray add(complexNumber2)
-	complexNumberArray add(complexNumber3)
+	this complexNumberArray add(this complexNumber0)
+	this complexNumberArray add(this complexNumber1)
+	this complexNumberArray add(this complexNumber2)
+	this complexNumberArray add(this complexNumber3)
```
Comment (davidpiuva) [inline test/math/FloatComplexVectorListTest.ooc]: Should void expressions with order dependent side effects be allowed in the declaration section?

Comment (shamanas) [inline test/math/FloatComplexVectorListTest.ooc]: For what it's worth, all rock does is place them in the `__defaults__` function in order, so the order of evaluation is guaranteed.  

Style wise, I wouldn't recommend it but that's a personal / per project thing :)

Comment (davidpiuva) [inline test/math/FloatComplexVectorListTest.ooc]: I suspected that and this code should probably to to a constructor or the same line as the declaration.

**Comment by davidpiuva** [comment]
@sebastianbaginski Peer review?

**Comment by davidpiuva** [comment]
Static calls were missed because the compiler resolved it as a namespace. More might come.

**Comment by davidpiuva** [comment]
Updated with missing `This` for the `expect` method.

**Comment by davidpiuva** [comment]
Updated tolerance.

**Pull request merged**